test_app/tests/lib/abstract_models/test_common.py

```diff
@@ -144,3 +144,34 @@ def test_cascade_behavior_for_created_by(user, user_api_client):
     org.refresh_from_db()
     assert org.created_by is None  # the SET_NULL behavior now implemented
     connection.check_constraints()
+
+
+@pytest.mark.xfail(reaason="https://github.com/ansible/django-ansible-base/issues/286")
+def test_do_not_update_modified_by_on_login(system_user, user, user_api_client):
+    user.refresh_from_db()
+    assert user.modified_by == system_user
+    with impersonate(user):
+        user_api_client.login(username=user.username, password='password')
+    user.refresh_from_db()
+    assert user.modified_by == system_user
+
+
+def test_modified_by_respects_given_value(system_user, random_user, user, animal):
+    animal.save()
+    assert animal.modified_by == system_user
+    with impersonate(user):
+        animal.modified_by = random_user
+        animal.save(update_fields=['modified_by'])
+    animal.refresh_from_db()
+    assert animal.modified_by == random_user
+
+
+def test_modified_by_gets_saved_even_if_not_in_update_fields(system_user, random_user, user, animal):
+    animal.save()
+    assert animal.modified_by == system_user
+    animal.name = 'Bob The Fish'
+    animal.kind = 'fish'
+    with impersonate(user):
+        animal.save(update_fields=['name', 'kind'])
+    animal.refresh_from_db()
+    assert animal.modified_by == user
```

Comment:
Contractually, I think this is _not_ what AWX did. IIRC, if you used update_fields but left out modified or modified_by, it wouldn't update those fields. I don't fully think I agree with that. I say we should match what the Django `auto_now` behavior is. Otherwise, no one will ever be able to predict how this will behave.